### PR TITLE
Feature/collisionhandling

### LINKED ISSUE #54: Handle collision rope - ball
The ball and rope should be removed if they collide, and the player should gain points.
If the ball is not size 1, the ball should split in two new balls with their size decreased.

### LINKED ISSUE #55: Handle collision ball - player
The player should lose a life if the ball hits the player. If the player has lost all his lives, the game should end

### PR DESCRIPTION
**Pull request by TMRert**
Full working game, with collision handlers for the ball and player and for the rope and the player. Fixes #54 and fixes #55

### CODE REVIEW

File: src/main/java/net/liquidpineapple/pang/objects/Ball.java [hunk 1]
```diff
@@ -88,15 +97,49 @@ public void move(){
...
+                hookInUse = true;
+                activeRope = (HookAndRope) o;
+            }
+        }
+
+        if(!hookInUse){
+            return false;
+        }
+
+        int ropePos = activeRope.getXPos() + (activeRope.getImage().getWidth(null))/2;
```
Comment (grappigegovert) [inline src/main/java/net/liquidpineapple/pang/objects/Ball.java]: Change `(activeRope.getImage().getWidth(null))` to `activeRope.getWidth()` ?

**Comment by grappigegovert** [comment]
Looking good, guys!

File: src/main/java/net/liquidpineapple/pang/objects/Ball.java (hunk 1 above)
Comment (TMRert) [inline src/main/java/net/liquidpineapple/pang/objects/Ball.java]: Yup that would be better indeed.

File: src/main/java/net/liquidpineapple/pang/objects/Player.java [hunk 2]
```diff
@@ -77,5 +80,31 @@ public void doUpdate() {
                 Application.getBoard().addObject(newRope);
             }
         }
+
+        if(collisionPlayer()){
+            if(!this.isHit) {
+                Application.lifeKeeper.loseLife();
+                this.isHit = true;
+            }
```
Comment (Fastjur) [inline src/main/java/net/liquidpineapple/pang/objects/Player.java]: These lines are all references to static methods and fields. It should thus be done like this:

```java
if(!isHit) {
    LifeSystem.loseLife();
    isHit = true;
}
```

File: src/main/java/net/liquidpineapple/pang/objects/LifeSystem.java [hunk 3]
```diff
@@ -38,21 +51,32 @@ public void updateLifes() {
...
     public static void gainLife() {
         if (lives < 3) {
             lives += 1;
         }
     }
 
+    /**
+     * Method that returns the current ammount of lives.
+     * @return - the remaining ammount of lives.
+     */
```
Comment (Fastjur) [inline src/main/java/net/liquidpineapple/pang/objects/LifeSystem.java]: Could you convert this whole method to the lombok `@Getter` on the field (I know you didn't write this method, but you're changing some things anyway)

File: src/main/java/net/liquidpineapple/pang/objects/Player.java [hunk 4]
```diff
@@ -77,5 +80,31 @@ public void doUpdate() {
...
+                int playerPos = this.getXPos() + (this.getImage().getWidth(null))/2;
+
+                if(playerPos - ball.getXPos() >= 0 && playerPos - ball.getXPos() <= ball.getWidth()) {
+                    if(ball.getYPos()+ ball.getHeight() >= this.getYPos()){
+                       return true;
+                    }
+                }
+            }
+        }
+        this.isHit = false;
```
Comment (Fastjur) [inline src/main/java/net/liquidpineapple/pang/objects/Player.java]: isHit is static. Remove the `this.`

**Comment by Wolgo** [comment]
I have taken the time to implement the small changes as indicated in the review because the deadline is closing in, don't want to risk the owners of this branch missing the review or forgetting to alter it in time.

**Pull request merged**